cloudiot_pubsub_example_mqtt_device.py

```diff
@@ -73,24 +79,46 @@ def error_str(rc):
 
 
 class Device(object):
-    """Represents the state of a single device."""
-
+    """Represents the state of a single device. Including the variables in the system."""
     def __init__(self):
+        #sensors
         self.temperature = 0
+        self.temp = temp
+        self.pH = 7
+        self.leak = 0
+        self.adc_sensors = adc.adc_sensors()
+        #self.adc.read_pH()
+        #self.adc.read_leak()
+        
         self.fan_on = False
         self.connected = False
+        
         self.led = LED(17)
         self.led.off()
+        
+        #Control Devices
+        self.peristaltic_pump_on = False
+        self.relay=relay
+        self.peristaltic_pump = self.relay
+        self.peristaltic_pump.init_one()
+        #self.peristaltic_pump.on1()
+        #self.peristaltic_pump.off1()
+
+        self.water_solenoid_on = False
+        self.water_solenoid = self.relay
+        self.water_solenoid.init_three()
+        #self.water_solenoid.on3()
+        #self.water_solenoid.off3()
 
     def update_sensor_data(self):
         """Pretend to read the device's sensor data.
         If the fan is on, assume the temperature decreased one degree,
         otherwise assume that it increased one degree.
         """
-        if self.fan_on:
-            self.temperature -= 1
-        else:
-            self.temperature += 1
+        self.temperature = temp.read()
```

Comment:
It would be really good to implement some error handling here! For instance, we want to recover if one of these reads messes up. We do not want the whole script to go down because it. Not sure what those possibilities are, but a try catch may be nice

Reply:
yeah this is a good point. I've done this for temp, but will do so for pH + leak too.